modules/smartyadsBidAdapter.js

```diff
@@ -79,19 +79,19 @@ export const spec = {
     let location;
     location = bidderRequest?.refererInfo ?? null;
     let placements = [];
+    
     let request = {
+      'placements': placements,
       'deviceWidth': winTop.screen.width,
       'deviceHeight': winTop.screen.height,
+      'host': location?.domain,
+      'page': location?.page,
       'language': (navigator && navigator.language) ? navigator.language : '',
-      'secure': 1,
-      'host': location?.domain ?? '',
-      'page': location?.page ?? '',
       'coppa': config.getConfig('coppa') === true ? 1 : 0,
```

Comment:
We will not be accepting tricks to fill the duplication check

Reply:
@patmmccann OK, but how can I fix that? I saw the report and duplication check marks, for example, the isBidResponseValid function is used in lots of adapters and looks almost the same.

Reply:
Check out this work in progress from an adapter structured very similar to yours

Reply:
https://github.com/MaksymTeqBlaze/Prebid.js/commit/0249eb7a258d610e937a114676e24c1d3f6c956e

Reply:
@patmmccann I'm sorry, but I still don't get it. The previous version of the isBidResponseValid function was flagged as duplicate code by your tool, but it doesn't need any updates, right? And the same function with same code is used in other adapters

Reply:
You and other sites have duplicate code, we need you to move common functions into libraries before making further updates to your adapter

Reply:
#8527 has an example on how to move common functions into a library